docs/docs/02-diagram_format.md

```diff
@@ -0,0 +1,376 @@
+---
+title: Diagram JSON Format
+---
+
+# JSON Format of `objects` argument in `draw` function
+
+> :warning: Dark mode may make example images harder to read.
+
+The `draw` function defined in `user_functions.ts` has an `objects` argument which is used to define the array of objects that is drawn on the canvas.
+One way to populate the `objects` argument is by using a JSON file containing the object array.
+
+## Core Attributes
+
+All drawable JSON objects include the following core attributes, unless explicitly stated otherwise:
+
+| Attribute | Type                | Required | Description                                                                                                                             |
+| --------- | ------------------- | -------- | --------------------------------------------------------------------------------------------------------------------------------------- |
```

Comment:
Use a colon (`:`) at the start of a `----` to indicate left-alignment. This will make the headers look better. See also https://www.markdownguide.org/extended-syntax/#alignment.